### PR TITLE
feat: welcome screen scrolls when keyboard is open

### PR DESCRIPTION
**Pull request by sracca**
This PR includes adapting the welcome screen so that it scrolls to show the text input box in the center of the screen when the keyboard is open, and resets itself when the keyboard is dismissed.
screen shot 2018-08-15 at 3 52 48 pm
screen shot 2018-08-20 at 1 26 40 pm 2

### CODE REVIEW

File: src/components/name-input.js [hunk 1]
```diff
@@ -10,18 +10,38 @@ import {
...
+    handleOnPress: PropTypes.func.isRequired,
+    isFocused: PropTypes.bool.isRequired,
+  };
 
-NameInput.propTypes = { nextPage: PropTypes.func.isRequired };
-
-export default NameInput;
+  render() {
+    const {
+      handleOnFocus,
```
Comment (geraldalewis) [inline src/components/name-input.js]: Use `on[Event]` to indicate the component is triggering an event; `handle[Event]` to handle it

File: src/containers/Welcome/index.js [hunk 2]
```diff
@@ -20,31 +26,58 @@ export default class Welcome extends React.Component {
     }).isRequired,
   };
 
+  state = {
+    inputMarginTop: 104,
+    isFocused: false,
+  };
+
+  scrollUp = () => {
```
Comment (geraldalewis) [inline src/containers/Welcome/index.js]: Let's rename this to `handleFocus` to better illustrate why the code is doing what it's doing. (Same for `scrollDown`).

File: src/containers/Welcome/index.js [hunk 3]
```diff
@@ -20,30 +26,58 @@ export default class Welcome extends React.Component {
...
+    this.setState({ inputMarginTop: 75, isFocused: true });
```
Comment (cbarber) [inline src/containers/Welcome/index.js]: `inputMarginTop` can be calculated from a ternary of `isFocused`. We can either pass both margins as `props` or maybe better to pass an offset that is conditionally subtracted.

File: src/components/name-input.js [hunk 4]
```diff
@@ -10,18 +10,34 @@ import {
...
-    <NextButton onPress={nextPage}>
-      <ButtonText>Next</ButtonText>
-      <WhiteArrowIcon source={ArrowIcon} />
-    </NextButton>
-  </NameInputView>
-);
+export default class NameInput extends React.Component {
+  static propTypes = {
+    handleOnFocus: PropTypes.func.isRequired,
+    handleMarginTop: PropTypes.number.isRequired,
```
Comment (cbarber) [inline src/components/name-input.js]: This name hints its a function (verb).

File: src/components/name-input.js [hunk 5]
```diff
@@ -1,25 +1,66 @@
...
+          />
+        }
+      </TextInputWrapper>
+      <NextButton onPress={handleOnPress}>
+        <ButtonText>
+          Next
+        </ButtonText>
+        <Image
+          /* eslint-disable-next-line */
+          source={ArrowIcon}
```
[the remark this answers is not in the document]
Comment (jennyckaplan) [reply src/components/name-input.js]: no clue it was already there when i pulled down the branch

**Pull request closed without merging**